app/routing/DeparturesRouter.scala

```diff
@@ -70,4 +72,22 @@ class DeparturesRouter @Inject() (
 
   }
 
+  def getMessageIds(departureId: String, receivedSince: Option[OffsetDateTime] = None): Action[Source[ByteString, _]] = route {
+    case Some(VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE) =>
+      (for {
+        convertedDepartureId <- implicitly[PathBindable[V2DepartureId]].bind("departureId", departureId)
+      } yield convertedDepartureId).fold(
+        bindingFailureAction(_),
+        converted => v2Departures.getMessageIds(converted, receivedSince)
+      )
+    case _ =>
+      (for {
+        convertedDepartureId <- implicitly[PathBindable[V1DepartureId]].bind("departureId", departureId)
+      } yield convertedDepartureId).fold(
+        bindingFailureAction(_),
+        converted => v1DepartureMessages.getDepartureMessages(converted, receivedSince)
+      )
```

Comment:
you can clean up some of the duplication here using something like this (or potentially provide the bindable name as a parameter if you want to reuse it for the arrival endpoints

`def runIfBound[A](action: A => Action) = 
implicitly[PathBindable[A]].bind("departureId", departureId).fold(bindingFailureAction(_), action)`